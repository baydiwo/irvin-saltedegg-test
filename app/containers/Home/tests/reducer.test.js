// import produce from 'immer';
import homeReducer from "../reducer";
import {
  FETCH_DETAIL_SUCCESS,
  ADD_TO_CART,
  REMOVE_ITEM,
  ADD_QUANTITY,
  TOTAL_PRICE,
} from "../actions";

/* eslint-disable default-case, no-param-reassign */
describe("homeReducer", () => {
  let state;
  beforeEach(() => {
    state = {
      products: [],
      addedItems: [],
      total: 0,
      tempTotal: 0,
    };
  });

  // it("returns the initial state", () => {
  //   const expectedResult = state;
  //   expect(homeReducer(undefined, {})).toEqual(expectedResult);
  // });
  it("FETCH_DETAIL_SUCCESS", () => {
    const action = {
      type: FETCH_DETAIL_SUCCESS,
      payload:  [
          {
            currency: "SGD",
            image:
              "http://s3-ap-southeast-1.amazonaws.com/s3.irvinsaltedegg.com/engineering-test/images/product-1.jpg",
            name: "IPhone 7",
            price: "15",
          },
        ],
    };
    expect(homeReducer(undefined, action)).toMatchSnapshot();
  });
  
  it("ADD_TO_CART", () => {
    const action = {
      type: ADD_TO_CART,
      payload:  [
          {
            currency: "SGD",
            image:
              "http://s3-ap-southeast-1.amazonaws.com/s3.irvinsaltedegg.com/engineering-test/images/product-1.jpg",
            name: "IPhone 7",
            price: "15",
          },
        ],
    };
    expect(homeReducer(undefined, action)).toMatchSnapshot();
  });
  
  it("REMOVE_ITEM", () => {
    const action = {
      type: REMOVE_ITEM,
      payload:  [
          {
            currency: "SGD",
            image:
              "http://s3-ap-southeast-1.amazonaws.com/s3.irvinsaltedegg.com/engineering-test/images/product-1.jpg",
            name: "IPhone 7",
            price: "15",
          },
        ],
    };
    expect(homeReducer(undefined, action)).toMatchSnapshot();
  });
  
  it("ADD_QUANTITY", () => {
    const action = {
      type: ADD_QUANTITY,
      payload:  [
          {
            currency: "SGD",
            image:
              "http://s3-ap-southeast-1.amazonaws.com/s3.irvinsaltedegg.com/engineering-test/images/product-1.jpg",
            name: "IPhone 7",
            price: "15",
          },
        ],
    };
    expect(homeReducer(undefined, action)).toMatchSnapshot();
  });
  
  it("TOTAL_PRICE", () => {
    const action = {
      type: TOTAL_PRICE,
      payload:  [
          {
            currency: "SGD",
            image:
              "http://s3-ap-southeast-1.amazonaws.com/s3.irvinsaltedegg.com/engineering-test/images/product-1.jpg",
            name: "IPhone 7",
            price: "15",
          },
        ],
    };
    expect(homeReducer(undefined, action)).toMatchSnapshot();
  });
  

  /**
   * Example state change comparison
   *
   * it('should handle the someAction action correctly', () => {
   *   const expectedResult = produce(state, draft => {
   *     draft.loading = true;
   *     draft.error = false;
   *     draft.userData.nested = false;
   *   });
   *
   *   expect(appReducer(state, someAction())).toEqual(expectedResult);
   * });
   */
});
