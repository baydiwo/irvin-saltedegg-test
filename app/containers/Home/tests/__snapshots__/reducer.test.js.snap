// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homeReducer ADD_QUANTITY 1`] = `
Object {
  "addedItems": Array [],
  "products": Array [],
  "tempTotal": 0,
  "total": 0,
}
`;

exports[`homeReducer ADD_TO_CART 1`] = `
Object {
  "addedItems": Array [],
  "products": Array [],
  "tempTotal": 0,
  "total": 0,
}
`;

exports[`homeReducer FETCH_DETAIL_SUCCESS 1`] = `
Object {
  "addedItems": Array [],
  "products": Array [],
  "tempTotal": 0,
  "total": 0,
}
`;

exports[`homeReducer REMOVE_ITEM 1`] = `
Object {
  "addedItems": Array [],
  "products": Array [],
  "tempTotal": 0,
  "total": 0,
}
`;

exports[`homeReducer TOTAL_PRICE 1`] = `
Object {
  "addedItems": Array [],
  "products": Array [],
  "tempTotal": 0,
  "total": 0,
}
`;
